File: medctrl-backend/API/api/models/medicine_models/procedure.py
# This program has been developed by students from the bachelor Computer Science at
# Utrecht University within the Software Project course.
# © Copyright Utrecht University (Department of Information and Computing Sciences)
from django.db import models
from .medicine import Medicine
from .lookupproceduretype import Lookupproceduretype


class Procedure(models.Model):
    """
    Model class for the procedure table. As the abstract functionality for all model classes is nearly identical, information about them has been moved to an external readme. 
    """
    eunumber = models.ForeignKey(Medicine, models.CASCADE, db_column="EUNumber")
    procedurecount = models.IntegerField(db_column="ProcedureCount")
    commisionnumber = models.IntegerField(
        db_column="CommisionNumber", blank=True, primary_key=True
    )
    emanumber = models.CharField(
        db_column="EMANumber", max_length=128, blank=True, null=True
    )
    proceduredate = models.DateField(db_column="ProcedureDate", blank=True, null=True)
    proceduretype = models.ForeignKey(
        Lookupproceduretype,
        models.CASCADE,
        db_column="ProcedureType",
        blank=True,
        null=True,
    )
    decisiondate = models.DateField(db_column="DecisionDate", blank=True, null=True)
    decisionnumber = models.CharField(
        db_column="DecisionNumber", max_length=45, blank=True, null=True
    )
    decisionurl = models.CharField(
        db_column="DecisionURL", max_length=255, blank=True, null=True
    )
    annexurl = models.CharField(
        db_column="AnnexURL", max_length=255, blank=True, null=True
    )

    manually_updated = models.BooleanField(default=False)

    class Meta:
        db_table = "procedure"
        unique_together = (("eunumber", "procedurecount"),)

File: medctrl-frontend/src/pages/detailed-info/components/Procedure.js
// This program has been developed by students from the bachelor Computer Science at
// Utrecht University within the Software Project course.
// © Copyright Utrecht University (Department of Information and Computing Sciences)
import ProcedureDetail from './ProcedureDetail'
import CustomLink from './CustomLink'

// Function based component, represents a single procedure and displays the
// following relevant information: the decision date and number, as well as the
// type of the procedure. There are also two links provided to both the annex
// and the decision file.
function Procedure(props) {
  const clean = (value) => {
    return !value ? 'NA' : value
  }

  let decisionDate = clean(props.proc.decisiondate)
  let procedureType = clean(props.proc.proceduretype)
  let decisionNumber = clean(props.proc.decisionnumber)

  let annexURL = props.proc.annexurl
  let decisionURL = props.proc.decisionurl

  return (
    <div className="med-info-procedure">
      <div className="med-info-procedure-content">
        <ProcedureDetail
          name="Decision Date"
          value={decisionDate}
          width="small-width"
        />
        <ProcedureDetail
          name="Procedure Type"
          value={procedureType}
          width="large-width"
        />
        <ProcedureDetail
          name="Decision Number"
          value={decisionNumber}
          width="small-width"
        />
      </div>

      <div className="med-info-procedure-file-link-container">
        <CustomLink
          className="med-info-procedure-file-link"
          name="Annex File"
          image="bx bx-file-blank"
          dest={annexURL}
        />

        <CustomLink
          className="med-info-procedure-file-link"
          name="Decision File"
          image="bx bx-file-blank"
          dest={decisionURL}
        />
      </div>
    </div>
  )
}

export default Procedure
